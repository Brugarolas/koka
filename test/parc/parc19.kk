fun test(xs:list<int>) {
  match(xs) {
    Cons(xs /* xs0 */) as ys -> /* dup xs0; drop xs */ xs /* xs0 */
    Nil -> /* drop xs */ 1
  }
}
