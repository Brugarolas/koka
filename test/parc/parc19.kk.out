public fun test : (xs : list<std/core/types/int>) -> std/core/types/int
 = fun(xs: list<std/core/types/int>){
 match (xs) {
 (std/core/Cons((xs0: std/core/types/int), (.pat: list<std/core/types/int>)) : list<std/core/types/int> )
 -> val _ : std/core/types/int
 = std/core/types/.dup(xs0);
 val _ : ()
 = std/core/types/.drop(xs);
 xs0;
 (std/core/Nil() : (list<std/core/types/int>) )
 -> std/core/types/.drop(xs);
 1;
 }
 };
test/parc/parc19.kk(3,10): warning: xs shadows an earlier local definition or parameter
test/parc/parc19.kk(3,27): warning: pattern variable ys is unused (or a wrongly spelled constructor?)
 hint: prepend an underscore to make it a wildcard pattern